data_structures/src/data_request.rs

```diff
@@ -338,48 +351,62 @@ pub fn create_tally<RT>(
     pkh: PublicKeyHash,
     report: &RadonReport<RT>,
     reveals: Vec<RevealTransaction>,
-) -> Result<TallyTransaction, RT::Error>
+) -> Result<TallyTransaction, failure::Error>
 where
     RT: TypeLike,
 {
-    let reveal_reward = calculate_dr_vt_reward(dr_output);
-    let n_reveals = reveals.len() as u16;
-
-    let mut outputs: Vec<ValueTransferOutput> = reveals
-        .into_iter()
-        .filter_map(|reveal| {
-            // Only reward reveals in consensus
-            if true_revealer(&reveal, report) {
-                let vt_output = ValueTransferOutput {
-                    pkh: reveal.body.pkh,
-                    value: reveal_reward,
-                    time_lock: 0,
-                };
-                Some(vt_output)
-            } else {
-                // TODO: Penalize dishonest nodes
-                None
+    if let Stage::Tally(tally_metadata) = report.metadata.clone() {
```

Comment:
You don't need this clone

```suggestion
    if let Stage::Tally(tally_metadata) = &report.metadata {
```